forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/test/java/com/forgerock/openbanking/aspsp/as/service/TppRegistrationServiceTest.java

```diff
@@ -0,0 +1,218 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.forgerock.openbanking.aspsp.as.service;
+
+import com.forgerock.openbanking.am.services.AMOIDCRegistrationService;
+import com.forgerock.openbanking.analytics.services.TppEntriesKPIService;
+import com.forgerock.openbanking.aspsp.as.api.registration.dynamic.dto.RegistrationError;
+import com.forgerock.openbanking.aspsp.as.configuration.ForgeRockDirectoryConfiguration;
+import com.forgerock.openbanking.aspsp.as.configuration.OpenBankingDirectoryConfiguration;
+import com.forgerock.openbanking.common.services.store.tpp.TppStoreService;
+import com.forgerock.openbanking.constants.OpenBankingConstants;
+import com.forgerock.openbanking.exceptions.OBErrorException;
+import com.forgerock.openbanking.jwt.services.CryptoApiClient;
+import com.forgerock.openbanking.model.oidc.OIDCRegistrationRequest;
+import com.nimbusds.jose.shaded.json.JSONObject;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.catchThrowableOfType;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TppRegistrationServiceTest {
+
+    CryptoApiClient cryptoApiClient = mock(CryptoApiClient.class);
+    OpenBankingDirectoryConfiguration openBankingDreictoryConfiguration = mock(OpenBankingDirectoryConfiguration.class);
+    ForgeRockDirectoryConfiguration forgeRockDirectoryConfiguration = mock(ForgeRockDirectoryConfiguration.class);
+    TppStoreService tppStoreService = mock(TppStoreService.class);
+    AMOIDCRegistrationService amoidcRegistrationService = mock(AMOIDCRegistrationService.class);
+    TppEntriesKPIService tppEntriesKPIService = mock(TppEntriesKPIService.class);
+    TppRegistrationService tppRegistrationService = null;
+
+    OIDCRegistrationRequest registrationRequest = mock(OIDCRegistrationRequest.class);
+    List<String> validRedirectUris = new ArrayList <>(List.of("https://google.co.uk"));
+    String softwareId = "SoftwareID";
+    JWTClaimsSet validSsaJwtClaims = null;
```

Comment:
Non-private variables! 😱 I know it doesn't matter in an isolated unit test, but I'm so used to the convention that it makes anything else more difficult to read! 😆

Reply:
Fixed